pystac/catalog.py

```diff
@@ -255,10 +255,12 @@ def add_child(
         child.set_root(self.get_root())
         if child.get_parent() is None or not keep_parent:
             child.set_parent(self)
+        elif keep_parent:
+            setattr(child, "keep_parent", True)
 
         # set self link
         self_href = self.get_self_href()
-        if self_href:
+        if self_href and not keep_parent:
```

Comment:
Shouldn't we update an self href if the child had no parent before?

```suggestion
        if self_href and not child.keep_parent:
```

Reply:
ah yes sorry I missed the difference there

Reply:
```suggestion
        if self_href and not getattr(child, "keep_parent", False):
```